tile_id_pass/tile_id_pass.py

```diff
@@ -0,0 +1,58 @@
+from gemstone.common.transform import replace, Generator, FromMagma
+from io_core.io_core_magma import IOCore
+from canal.interconnect import Interconnect
+from gemstone.common.configurable import Configurable, ConfigurationType
+import magma
+import mantle
+from gemstone.generator.const import Const
+from hwtypes import BitVector
+
+## We create constant tie hi and tie lo outputs on each tile
+## These pins will be interleaved with the tile_id pins as follows
+## | hi[0] | id[0] | lo[0] | id[1] | hi[1] | id[2] | lo[1] | id[3] | hi[2] | ...
+## We are doing this so that we can actually connect the tile_id inputs
+## to the proper constant values without uniquifying the tiles and while
+## maintaining our abutted array of tile hard macros
+def tile_id_physical(interconnect: Interconnect):
+    tile_id_width = interconnect.tile_id_width
+    tie_hi_width = (tile_id_width // 2) + 1
```

Comment:
Shouldn't this be `ie_hi_width = int(math.ceil(tile_id_width / 2))`?

Say `tile_id_width = 16`, based on the original code, `tile_hi_width` would be 9, so you will create a 17-bit signal in the end.

Reply:
No, I actually want 17 bits here. Every tile_id bit must have both a hi output and a lo output next to it. To do this, you need one more tied output than tile_id input pins. It's clear when you look at the interleaving pattern like this:

| hi | id | lo | id | hi | 

2 id pins, 1 lo, 2 hi (3 ties in total)